src/components/SearchInput/SearchInput.test.tsx

```diff
@@ -0,0 +1,181 @@
+import { render, screen, waitFor } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { act } from "react-dom/test-utils"
+import { SearchInput } from "./SearchInput"
+
+describe("SearchInput", () => {
+  it("renders search icon", () => {
+    // Arrange
+    const searchIconTestId = "search_icon"
+
+    // Act
+    render(<SearchInput onDebounce={jest.fn()} />)
+
+    const receivedIcon = screen.getByTestId(searchIconTestId)
+
+    // Assert
+    expect(receivedIcon).toBeInTheDocument()
+  })
+  it("clear icon is not visible if search term is empty", () => {
+    // Arrange
+    const label = "click icon to clear search"
+
+    // Act
+    render(<SearchInput onDebounce={jest.fn()} />)
+
+    const receivedIcon = screen.getByLabelText(label)
+
+    // Assert
+    expect(receivedIcon).not.toBeVisible()
+  })
+  it("debounces and invokes a callback", async () => {
+    // Arrange
+    jest.useFakeTimers()
+    const user = userEvent.setup({ delay: null })
+    const onDebounce = jest.fn()
+    const label = "search"
+    const text = "hello world"
+    const debounceDelay = 600
+
+    // Act
+    render(
+      <SearchInput
+        aria-label={label}
+        onDebounce={onDebounce}
+        debounceDelay={debounceDelay}
+      />,
+    )
+
+    const input = screen.getByLabelText(label)
+
+    await user.type(input, text)
+
+    act(() => {
+      jest.advanceTimersByTime(debounceDelay)
+    })
+
+    // Assert
+    await waitFor(() => expect(onDebounce).toBeCalled())
+    jest.useRealTimers()
+  })
+  it("has a default delay of 300", async () => {
+    // Arrange
+    jest.useFakeTimers()
+    const user = userEvent.setup({ delay: null })
+    const onDebounce = jest.fn()
+    const label = "search"
+    const text = "hello world"
+
+    // Act
+    render(<SearchInput aria-label={label} onDebounce={onDebounce} />)
+
+    const input = screen.getByLabelText(label)
+
+    await user.type(input, text)
+
+    act(() => {
+      jest.advanceTimersByTime(300)
+    })
+
+    // Assert
+    await waitFor(() => expect(onDebounce).toBeCalled())
```

Comment:
nit ⛏️ : this test would pass if the default delay was any number less than 300. You could add an expect before the `waitFor` to assert that it hasn't been fired before the debounce time.